test_requests.py

```diff
@@ -1261,6 +1262,20 @@ def test_preserve_last_key_case(self):
         assert frozenset(cid.keys()) == keyset
         assert frozenset(cid) == keyset
 
+    def test_copy(self):
+        cid = CaseInsensitiveDict({
+            'Accept': 'application/json',
+            'user-Agent': 'requests',
+        })
+        assert cid == cid.copy()
```

Comment:
Should this test also assert that the copy does not lead to two objects that are literally the same? e.g., should we also add `assert cid is not cid.copy()`?